.github/workflows/test-and-publish.yaml

```diff
@@ -53,3 +50,9 @@ jobs:
             - name: run tests on windows/mac
               run: npm test
               if: runner.os != 'Linux'
+
+            - name: Publish to extension marketplace when new tag is pushed
```

Comment:
do we need to limit this to the main branch?

Reply:
This step only runs if the run was triggered by a tag being pushed. We could check that the tagged commit is on the master branch ([here are some ways to do that](https://stackoverflow.com/a/2707110)) or trust ourselves to only tag commits on master.

I'll experiment with this tonight.